lib/reducers/app_reducer.dart

```diff
@@ -72,8 +73,11 @@ class StoreProcessingFailuresReducer
 
 class AddProblemReducer extends TypedReducer<AppState, AddProblem> {
   AddProblemReducer()
-      : super((state, action) =>
-            state.rebuild((b) => b..problems.add(action.problem)));
+      : super((state, action) => state.rebuild((b) => b
```

Comment:
Cool, so we're adding a ProblemPageData with each new problem.  Nice place to start hey... looks good!